### PR TITLE
[CDAP-7173] - Initial implementation of new Login webpack + integration

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds login webapp in UI
- Integrates Hydrator & Tracker with login flow in UI
- Removes un-necessary login functionality from within Hydrator & Tracker
- Adds build process for UI to be included in the SDK 
- Adds preliminary addition of eslint support for advanced es7 syntax. I am not using es7 syntax in this PR yet but it will slowly start to come. It is not fully functional yet as there are some issues on es7 syntax usage as of now. We will have to see if we need to remove it from this PR for later use.
#### Todo:
- Add encryption to token in client side cookies.
#### Note:
- Have to confirm on the build process. I wouldn't worry about the changes for now until we have a easy solution to use old UI in the cluster environment and use new UI only in local sdk.

### CODE REVIEW

File: cdap-ui/.eslintrc.json [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
+    "babel/no-await-in-loop": 1,
+    "babel/flow-object-type": 1,
+    "babel/func-params-comma-dangle": 1
   },
   "env": {
     "browser": true,
     "commonjs": true,
-    "node": true
+    "node": true,
+    "es6": true
```
Comment (elfenheart) [inline cdap-ui/.eslintrc.json:29]: not 7? the parser is using ecma7?

File: cdap-ui/app/directives/my-global-navbar/my-global-navbar.js [hunk 2]
```diff
@@ -22,6 +22,18 @@ function NavbarController ($scope, $state, myNamespace, EventPipe, MYAUTH_EVENT,
   vm.$cookies = $cookies;
 
   function findActiveProduct() {
+    var baseTag, baseUrl;
+    if ($state.is('userprofile')) {
+      baseTag = document.getElementsByTagName('base');
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-global-navbar/my-global-navbar.js]: does this work? Is there a <base> element?

File: cdap-ui/app/login/components/Card/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    this.props = props;
+  }
+  render() {
+    var statusClasses = classNames('card-status',
+      { 'error' : this.props.error },
+      { 'success': this.props.success },
+      { 'hide': !this.props.error && !this.props.success }
+    );
+    return (
+      <Card-Wrapper>
```
Comment (elfenheart) [inline cdap-ui/app/login/components/Card/index.js]: Are we still going to go with this? I think React still trying to look for this class

File: cdap-ui/app/login/login.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                      Remember me
+                    </label>
+                  </div>
+                </div>
+                <button
+                  type="submit"
+                  className="btn btn-primary pull-right"
+                  disabled={!this.state.formState}
+                  onClick={this.login.bind(this)}
+                  >
```
Comment (elfenheart) [inline cdap-ui/app/login/login.js]: nit: unindent by 1

File: cdap-ui/.eslintrc.json (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/.eslintrc.json:29]: Talked to @elfenheart . There is no es7 environment currently available in eslint.

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**